webtau-http/src/main/java/com/twosigma/webtau/http/validation/HeaderDataNode.java

```diff
@@ -17,19 +17,52 @@
 package com.twosigma.webtau.http.validation;
 
 import com.twosigma.webtau.data.traceable.TraceableValue;
+import com.twosigma.webtau.http.HttpResponse;
 import com.twosigma.webtau.http.datanode.DataNode;
+import com.twosigma.webtau.http.datanode.DataNodeBuilder;
 import com.twosigma.webtau.http.datanode.DataNodeId;
 import com.twosigma.webtau.http.datanode.NullDataNode;
 
+import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class HeaderDataNode implements DataNode {
+    private static final Set<CamelCaseTranslation> translations = setOf(
+            new CamelCaseTranslation("Content-Location", "contentLocation"),
+            new CamelCaseTranslation("Content-Length", "contentLength", Integer::valueOf),
+            new CamelCaseTranslation("Content-Encoding", "contentEncoding")
+    );
+
     private final DataNode dataNode;
 
-    public HeaderDataNode(DataNode dataNode) {
-        this.dataNode = dataNode;
+    public HeaderDataNode(HttpResponse response) {
+        Map<String, Object> headerData = new HashMap<>();
+
+        headerData.put("statusCode", response.getStatusCode());
+        headerData.put("contentType", response.getContentType());
+
+        response.getHeader().forEachProperty(headerData::put);
+
+        translations.forEach(translation -> addCamelCaseVersion(headerData, translation));
+
+        this.dataNode = DataNodeBuilder.fromMap(new DataNodeId("header"), headerData);
+    }
+
+    private static void addCamelCaseVersion(Map<String, Object> headerData, CamelCaseTranslation translation) {
```

Comment:
I realise that doing this for each header we're trying to translate isn't terribly efficient, essentially O(n*m) where n is the number of translations and m the number of headers.  However, for any realistic number of headers and translations this is not going to be an issue at all.  If we feel it would, I can implement this in a different way but it would be a little less obvious.